### ISSUE TITLE
Protege improperly renders IRIs with colons

### ISSUE THREAD

**Issue by swartik** [protegeproject/protege#721]
I'm using an ontology with individuals whose IRIs have embedded colons. These individuals denote times. An example would be http://www.ida.org/colon-iris#t05:38.

Protege does not render this IRI correctly. Assuming prefix p is http://www.ida.org/colon-iris#, Protege displays just the prefix, "p:". Here's a simplified version of the ontology:

    @prefix p: <http://www.ida.org/colon-iris#> .
    @prefix owl: <http://www.w3.org/2002/07/owl#> .
    @prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
    @base <http://www.ida.org/colon-iris> .
    <http://www.ida.org/colon-iris> rdf:type owl:Ontology .
    p:TimeOfDay rdf:type owl:Class .
    p:t05:38 rdf:type p:TimeOfDay .

Look at the individuals, and you'll see that t05:38 isn't rendered correctly:

![screenshot](https://user-images.githubusercontent.com/6043991/31282695-59ac707e-aa81-11e7-974d-057b6227940e.png)

I created this example using Protege. When I saved the ontology in Turtle format, the individual's IRI was just "p:". That is, the last line was:

    p: rdf:type p:TimeOfDay .

Is this also an OWLAPI issue?

**Comment by matthewhorridge** [comment]
Hi Steve,

It could be an OWL API issue.  I need to investigate.  Thanks for the report.

**Comment by swartik** [comment]
Matthew,

I just noticed that Protege (or whoever) has a similar problem with #'s. That made me wonder if all the characters in Turtle [reserved character escapes](https://www.w3.org/TR/2014/REC-turtle-20140225/#sec-escapes) were problematic. The answer is: partly.

Here is a screenshot showing what happens when you enter every reserved character through Protege. In each case, I entered the IRI wcw, where "w" is a word (amp, at, bang, etc.) and "c" is the character (&, @, !). I entered colon, percent, and sharp using %XX format.

![individuals](https://user-images.githubusercontent.com/6043991/31620813-3ebcbc5a-b266-11e7-8582-92086feeb624.png)

Only hyphen, dot, and underscore display properly. In the other cases, Protege renders only the word following the reserved character. To enter the individual with a colon, I used the sequence "colon%3Acolon" and for some reason Protege renders the "A".

The comments in the Turtle file show the full IRI, but the actual IRI omits the first word and the reserved character. For instance:

    ### http://www.ida.org/singletons#dollar$dollar
    s:dollar rdf:type owl:NamedIndividual .

The comments for the names I entered using %XX sequences show %XX.

Next, I tried loading an ontology with local names containing reserved escapes:

    s:amp\&amp rdf:type owl:NamedIndividual .

Protege rejected each such line. I'm guessing this is an OWL API issue.

Steve Wartik

**Comment by matthewhorridge** [comment]
Thanks for this Steve.  Much appreciated.  I think I'm going to have to write a new short form provider.  The OWL API has become more specific that what we want in Protege.

**Comment by barrynl** [comment]
This issue might be related?

When I create an ontology in Protege 5.2.0 with IRI: `http://192.168.99.100/pizza/` and create an individual with the following IRI: `http://192.168.99.100/pizza/7-Vulcano`, it will be saved as Turtle as follows (note that the `7-` of the individual name gets removed):

```
@prefix : <http://192.168.99.100/pizza/> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@base <http://192.168.99.100/pizza/> .

<http://192.168.99.100/pizza/> rdf:type owl:Ontology .

#################################################################
#    Individuals
#################################################################

###  http://192.168.99.100/pizza/7-Vulcano
:Vulcano rdf:type owl:NamedIndividual .


###  Generated by the OWL API (version 4.2.8.20170104-2310) https://github.com/owlcs/owlapi
```

**Comment by matthewhorridge** [comment]
I believe that this issue is fixed.  If this is not the case please comment, otherwise we'll close the issue after a couple of weeks.

**Comment by matthewhorridge** [comment]
Thanks @swartik, I appreciate it.  I need to investigate further...

**Comment by matthewhorridge** [comment]
Issues filed on the OWL API tracker:  https://github.com/owlcs/owlapi/issues/817

**Comment by matthewhorridge** [comment]
I'm going to push this back to the 5.6.0 milestone, because it's going to block the 5.5.0 milestone otherwise.

**Comment by matthewhorridge** [comment]
Fixed by upgrading the OWL API to 4.5.10
